add spec for $LOAD_PATH.resolve_feature_path

lxxxvi opened the pull request:
Hello 👋 

Here's my draft for `$LOAD_PATH.resolve_feature_path`.
I am quite certain that the specs are in the wrong place, can you tell me where to put it?

Thanks

eregon commented:
> can you tell me where to put it?

It's a good question.
I think either `language/predefined_spec.rb` or `core/kernel/load_path_spec.rb` would be good places.
I have a slight preference for `language/predefined_spec.rb` since it's a really a singleton method of that global variable.

eregon commented on core/kernel/shared/require.rb:
```diff
@@ -774,3 +774,24 @@
+    it "returns what will be loaded without actual loading, .so file" do
+      extension, path = $LOAD_PATH.resolve_feature_path('etc')
+      extension.should == :so
+      path.should =~ /\/etc\.(so|bundle)$/
```
Could you use `.end_with?("etc.#{RbConfig::CONFIG['DLEXT']}")`? (and add `require 'rbconfig` at the start of the `it`)

eregon commented on core/kernel/shared/require.rb:
```diff
@@ -774,3 +774,24 @@
+    end
+
+    # this may change (?!) https://bugs.ruby-lang.org/issues/16043
+    it "raises if feature cannot be found" do
```
Minor: `raises LoadError if ...`

lxxxvi commented:
@eregon Thanks for your feedback.

I adjusted the specs according to your input and moved them to the bottom of `language/predefined_spec.rb`. 

Let me know if you have any additional input. 🙌

eregon reviewed (approved):
Thank you for the new specs, looks great!
